src/Core/Context.idr

```diff
@@ -585,9 +585,11 @@ HasNames Def where
       fullNamesPat (_ ** (env, lhs, rhs))
           = pure $ (_ ** (!(full gam env),
                           !(full gam lhs), !(full gam rhs)))
-  full gam (TCon t a ps ds u ms cs det)
+  full gam (TCon t a ps ds u ms Nothing det)
+      = pure $ TCon t a ps ds u !(traverse (full gam) ms) Nothing det
+  full gam (TCon t a ps ds u ms (Just cs) det)
       = pure $ TCon t a ps ds u !(traverse (full gam) ms)
-                                !(traverse (full gam) cs) det
+                                (Just !(traverse (full gam) cs)) det
```

Comment:
Should this be
```idris
  full gam (TCon t a ps ds u ms mcs det)
      = pure $ TCon t a ps ds u !(traverse (traverse (full gam)) mcs) Nothing det
```
?

Reply:
Yeah, I was trying to think of the which function did that and it was sitting there right in front of me. I'll take care of these after work.

Reply:
But it turns out that `Core` is special and weird, so we need a `traverseOpt`:
```idris
  full gam (TCon t a ps ds u ms mcs det)
      = pure $ TCon t a ps ds u !(traverse (full gam) ms)
                                !(traverseOpt (traverse (full gam)) mcs) det
```